doc: move puncover down on the page.

Flameeyes opened the pull request:
doc: move puncover down on the page.


The puncover tool does not appear to be maintained anymore and it is
affected by multiple issues. Providing it as the first in the list
is a disservice to users as they might get frustrated and not notice
the `ram_report` tool that actually works.

Move the information at the bottom of the section, and add a warning
about the fact that this is a third party tool that might or might
not be working.

(Also make the link consistent to avoid "here" being a hyperlink, it
adds nothing to the reader either in RST or HTML.)

Signed-off-by: Diego Elio Pettenò <flameeyes@meta.com>

carlescufi reviewed (commented):
Looks good, but can you take the opportunity to add a note about the project not being that active anymore perhaps?

kartben commented on doc/develop/optimizations/tools.rst:
```diff
@@ -156,6 +124,42 @@ which will generate something similar to the output below::
+Build Target: puncover
+======================
+
+This target uses a third party tools called puncover which can be found at
```
```suggestion
This target uses a third-party tool called puncover which can be found at
```

kartben commented on doc/develop/optimizations/tools.rst:
```diff
@@ -156,6 +124,42 @@ which will generate something similar to the output below::
+
+.. warning::
+
+   This is a third party tool that might or might not be working at any given
```
```suggestion
   This is a third-party tool that might or might not be working at any given
```

kartben commented on doc/develop/optimizations/tools.rst:
```diff
@@ -156,6 +124,42 @@ which will generate something similar to the output below::
+This target uses a third party tools called puncover which can be found at
+https://github.com/HBehrens/puncover. When this target is built, it will
+launch a local web server which will allow you to open a web client and browse
+the files and view their ROM, RAM and stack usage. Before you can use this
```
```suggestion
the files and view their ROM, RAM, and stack usage. Before you can use this
```

Flameeyes committed:
doc: move puncover down on the page.


The puncover tool does not appear to be maintained anymore and it is
affected by multiple issues. Providing it as the first in the list
is a disservice to users as they might get frustrated and not notice
the `ram_report` tool that actually works.

Move the information at the bottom of the section, and add a warning
about the fact that this is a third party tool that might or might
not be working.

(Also make the link consistent to avoid "here" being a hyperlink, it
adds nothing to the reader either in RST or HTML.)

Signed-off-by: Diego Elio Pettenò <flameeyes@meta.com>

Flameeyes reviewed (commented):
Applied suggestions.
